[Interop] WPT test "pointerevent_click_is_a_pointerevent.html?mouse" is failing when run via webdriver

burg opened the pull request:
#### f51e778746b77dd1a3feb64cf747e8bdb8462e5c
<pre>
[Interop] WPT test &quot;pointerevent_click_is_a_pointerevent.html?mouse&quot; is failing when run via webdriver
<a href="https://bugs.webkit.org/show_bug.cgi?id=298676">https://bugs.webkit.org/show_bug.cgi?id=298676</a>

Reviewed by Lily Spiniolas.

Build on the previous attempt by adding a special case for open dialogs.

WebAutomationSessionProxy::focusFrame now takes an optional frameID instead of
a frameID, and focuses the main frame if frameID is not set. The method is
now called by WebAutomationSession::switchToBrowsingContext even when frameID
is empty so that focus is explicitly returned to the main frame.

* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebKit/CMakeLists.txt:
* Source/WebKit/Shared/Automation/WebAutomationSessionMacros.h:
Renamed from Source/WebKit/UIProcess/Automation/WebAutomationSessionMacros.h.
Move this header to WebKit/Shared/ so it can be included from WebProcess code.

* Source/WebKit/UIProcess/Automation/WebAutomationSession.cpp:
(WebKit::WebAutomationSession::switchToBrowsingContext):
Don&apos;t try to focus frame if the page is showing a dialog. If the page
is showing a modal dialog, it won&apos;t handle any incoming messages
and the call to FocusFrame will deadlock, causing a timeout.

* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp:
(WebKit::WebAutomationSessionProxy::focusFrame):
Rewrite this code more succinctly using ASYNC_FAIL_WITH_PREDEFINED_ERROR_IF.

* Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.h:
* Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.messages.in:

Canonical link: <a href="https://commits.webkit.org/300711@main">https://commits.webkit.org/300711@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/385432e6d93fe231787fb0a2c16ebd13ea9c2a80

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/123590 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/43305 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/34001 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/130357 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/75733 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/992bf121-f457-49b5-97f7-8d485b2f83b4) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/125467 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/44028 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/51899 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/93983 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/62374 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/ca1b37d1-06f8-4f1b-9432-18649d4d1c89) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/126543 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/35075 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/110569 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/74575 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/9404b0fb-1f81-48dc-8ca6-0edd3e1dc56e) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/34043 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/28727 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/73836 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/104794 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/28951 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/133050 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/50541 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/38481 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/102446 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/50916 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/106790 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/102288 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25999 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/47638 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/25882 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/47352 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/50395 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/56157 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/49869 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/53216 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/51544 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/6b9bd9635ee853f56132e542eb5cfffffa8dc142. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51511

burg commented on Source/WebKit/UIProcess/Automation/WebAutomationSession.cpp:
```diff
@@ -485,11 +485,14 @@ void WebAutomationSession::switchToBrowsingContext(const Inspector::Protocol::Au
+            return;
+        }
+
+        page->sendWithAsyncReplyToProcessContainingFrameWithoutDestinationIdentifier(frameID, Messages::WebAutomationSessionProxy::FocusFrame(page->webPageIDInMainFrameProcess(), frameID.value()), WTF::CompletionHandler<void(Inspector::CommandResult<void>&&)> { [callback = WTFMove(callback)] (auto result) mutable {
```
The result is Expected<String> so  i don't think it matters.

burg committed:
[Interop] WPT test "pointerevent_click_is_a_pointerevent.html?mouse" is failing when run via webdriver
https://bugs.webkit.org/show_bug.cgi?id=298676

Reviewed by Lily Spiniolas.

Build on the previous attempt by adding a special case for open dialogs.

WebAutomationSessionProxy::focusFrame now takes an optional frameID instead of
a frameID, and focuses the main frame if frameID is not set. The method is
now called by WebAutomationSession::switchToBrowsingContext even when frameID
is empty so that focus is explicitly returned to the main frame.

* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebKit/CMakeLists.txt:
* Source/WebKit/Shared/Automation/WebAutomationSessionMacros.h:
Renamed from Source/WebKit/UIProcess/Automation/WebAutomationSessionMacros.h.
Move this header to WebKit/Shared/ so it can be included from WebProcess code.

* Source/WebKit/UIProcess/Automation/WebAutomationSession.cpp:
(WebKit::WebAutomationSession::switchToBrowsingContext):
Don't try to focus frame if the page is showing a dialog. If the page
is showing a modal dialog, it won't handle any incoming messages
and the call to FocusFrame will deadlock, causing a timeout.

* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp:
(WebKit::WebAutomationSessionProxy::focusFrame):
Rewrite this code more succinctly using ASYNC_FAIL_WITH_PREDEFINED_ERROR_IF.

* Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.h:
* Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.messages.in:

Canonical link: https://commits.webkit.org/300711@main
